### PR TITLE
feat: improving build.rs, Solidity scripts and Just tasks

### PR DESCRIPTION
**Pull request by dinhani-cw**
* Created bash scripts to clone, compile and execute tests on Solidity contracts.
* Organized Just tasks by context (check their descriptions).
* Added Just task to setup the project after a checkout.
* Making E2E tests more reliable by using `wait-service` and `killport` utilities.
* Fixed a bug where the input data may not be present in the payload. In that case, asssumes it is empty.
* Changed build.rs to automatically include all signatgures files present in the project.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -18,22 +19,26 @@ fn main() {
...
 // Solidity signatures
 // -----------------------------------------------------------------------------
-const SIGNATURE_FILES: [&str; 5] = [
-    "static/contracts/BRLCToken.signatures",
-    "static/contracts/CardPaymentProcessor.signatures",
-    "static/contracts/CashbackDistributor.signatures",
-    "static/contracts/PixCashier.signatures",
-    "static/contracts/TokenDistributor.signatures",
-];
+const SIGNATURES_DIR: &str = "static/contracts/*.signatures";
```
Comment (dinhani-cw) [inline build.rs:22]: Automatically loading all signatures files in the specified folder, so we can have the function names in logs and metrics.

File: src/eth/primitives/call_input.rs [hunk 2]
```diff
@@ -5,5 +5,6 @@ use crate::eth::primitives::Bytes;
 pub struct CallInput {
     pub from: Address,
     pub to: Option<Address>,
+    #[serde(default)]
```
Comment (dinhani-cw) [inline src/eth/primitives/call_input.rs:8]: This field may not be present in the request payload, so we assume empty as default.

File: justfile [hunk 3]
```diff
@@ -1,61 +1,85 @@
-# Runs the service locally
+# Project: Show available tasks
+default:
+    just --list --unsorted
+
+# Project: Run project setup
+setup:
```
Comment (dinhani-cw) [inline justfile:6]: Setup task that automatically configures some additional tools.

File: justfile [hunk 4]
```diff
@@ -1,61 +1,85 @@
-# Runs the service locally
+# Project: Show available tasks
+default:
+    just --list --unsorted
+
+# Project: Run project setup
```
Comment (dinhani-cw) [inline justfile:5]: Setup task that automatically configures some additional tools.

File: justfile [hunk 5]
```diff
@@ -123,10 +139,35 @@ e2e-ledger:
...
+# E2E: Lint and format code
+e2e-lint:
+    #!/bin/bash
+    if [ -d e2e ]; then
+        cd e2e
+    fi
+    node_modules/.bin/prettier . --write
+
+# ------------------------------------------------------------------------------
+# Contracts tasks
```
Comment (dinhani-cw) [inline justfile:159]: Solidity contracts tasks

**Review by arthurmm-cloudwalk — APPROVED** [review]

**Pull request merged**